transport/grpc/errors.go

```diff
@@ -0,0 +1,25 @@
+package grpc
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
+)
+
+type errCantExtractHeader struct {
```

Comment:
nit: rename to errCannotExtractHeader. Contractions don't really work without the `'`.